### ISSUE TITLE
GPU doesn't seem to work

### ISSUE THREAD

**Issue by Mozoloa** [C0untFloyd/roop-unleashed#122]
**Describe the bug**
I've done a clean install, installed cuda toolkit & cudnn correctly according to roop documentation (btw your link to it is dead)
But it doesn't seem to work on my machine (3090 windows 11), I've also ran
```
pip uninstall onnxruntime onnxruntime-gpu
pip install onnxruntime-gpu==1.15.1
```
just in case but the gen is stuck at just a few frames/s when it should be around 20 in my past experience. Cuda is selected in the settings as well, reselected it and restarted but it's still not working.

I'm using the latest studio drivers for 3090

Sometimes it seems to go very slow then very fast towards the end, somethings isn't right

**To Reproduce**
1. Clean install
2. Install Cuda 11.8 + cudNN
3. Make sure cuda is selected in the settings
4. Try to roop an image
5. Observe as it's probably not using the GPU

**Details**
What OS are you using?
- [ ] Linux
- [ ] Linux in WSL
- [ x] Windows 
- [ ] Mac

Are you using a GPU?
- [ ] No. CPU FTW
- [ x] NVIDIA
- [ ] AMD
- [ ] Intel
- [ ] Mac

**Which version of roop unleashed are you using?**

2.7.3

**Screenshots**
![image](https://github.com/C0untFloyd/roop-unleashed/assets/20817233/47fb84f3-e894-46fc-928c-78cff4ec04be)

**Comment by Mozoloa** [comment]
Okay I just realised I was using codeformer post processing, which greatly slowed down the gen (10x slower) so that must have been it, but is this normal ?

**Comment by eniora** [comment]
Ye codeformer is very slow, I only use GFPGAN for enhancing the faces. BTW I think installing Cuda 11.8 + cudNN isn't necessary for unleashed since torch is imported here unlike with the latest original roop.

**Comment by Gasbeutel** [comment]
Using the mask feature seems to switch off GPU usage as well. It gets very slow and GPU usage in task manager stays at 0%...

**Comment by Mozoloa** [comment]
> Ye codeformer is very slow, I only use GFPGAN for enhancing the faces. BTW installing Cuda 11.8 + cudNN isn't necessary for unleashed since torch is imported here unlike with the latest original roop which removed the **import torch** line in v1.3.2.

I've tried gfpgan and it seems even slower, but it might be normal

**Comment by lysxelapsed** [comment]
swapping with an enhancer enabled will slow down speed drastically. so that's normal, yes. also reduce the number of threads when using an enhancer, see here: #116
to check gpu usage, try "gpu-z", which is a free and lightweight gpu-tool. the windows task manager often doesn't show the usage accurately in my experience.

**Comment by katherineex** [comment]
Yes i have the same issue the graphic card seem to not working it slow as hell
![image](https://github.com/C0untFloyd/roop-unleashed/assets/53106784/11e4f5a0-5024-4a86-9f85-0510c8ed44b1)

**Comment by C0untFloyd** [comment]
#103

**Comment by pablo1794** [comment]
why is the latest so slow on the latest alienware machines windows 11 ? GTX 4060 BPU

**Comment by C0untFloyd** [comment]
https://github.com/C0untFloyd/roop-unleashed/wiki/FAQ#why-is-my-graphics-card-gpu-not--hardly-being-utilized

**Comment by sahad1001** [comment]
Y

**Comment by sahad1001** [comment]
Make a photo
